community/learn/graphql-tutorials/tutorials/typescript-react-apollo/tutorial-site/content/mutations-variables/3-create-mutation.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "Run mutation, update cache"
+metaTitle: "Apollo Mutation with useMutation Hook | GraphQL React Apollo Typescript Tutorial"
+metaDescription: "We will use the Apollo useMutation hook from @apollo/react-hooks in React app as an example to insert new data and update cache locally using readQuery and writeQuery."
+---
+
+### Apollo useMutation Hook
+Now let's do the integration part. Open `src/components/Todo/TodoInput.tsx` and add the following code below the other imports:
+
+```javscript
+import { useMutation } from "@apollo/react-hooks";
+```
+
+We are importing the `useMutation` hook from `@apollo/react-hooks` and passing it the `ADD_TODO` mutation that we defined in the previous step.
+
+```javascript
+
+  const TodoInput = ({isPublic=false}) => {
+    const [todoInput, setTodoInput] = React.useState('');
++   const [addTodo] = useMutation(ADD_TODO);
+    return (
+
+```
+
+In the `useMutation` hook defined above, the first property of the result object is the mutate function; (addTodo) in this case. Read more about the mutate function [here](https://www.apollographql.com/docs/react/essentials/mutations.html)
+
+The mutate function optionally takes variables, optimisticResponse, refetchQueries, and update; You are going to make use of the `update` function later.
```

Comment:
Talk about `refetchQueries` and `optimisticResponse` later when you are actually using them.
This is the first place where you are introducing the mutate function. The other things would just confuse the reader. Keep it limited to `mutate` and `variables`.